src/ast/codegen_llvm.cpp

```diff
@@ -1004,6 +1004,24 @@ void CodegenLLVM::visit(Call &call)
     auto arg = call.vargs->at(0);
     auto scoped_del = accept(arg);
   }
+  else if (call.func == "macaddr")
+  {
+    // Mac addresses are presented as char[6]
+    AllocaInst *buf = b_.CreateAllocaBPF(b_.getInt64Ty(), "macaddr");
```

Comment:
```suggestion
    AllocaInst *buf = b_.CreateAllocaBPFInit(call.type, "macaddr");
```

You create a i64 type with this, storing that in an `[6*i8]` later causes issues. `IRBuilderBPF::GetType` knows how to handle array types so it shoudl work ok

Reply:
Thanks. The latest commit has this change, but I'm still seeing this in my local build:

```
4: /usr/bin/llvm-as-7: /vagrant/tests/codegen/llvm/call_macaddr.ll:28:9: error: stored value and pointer type do not match
4:   store [6 x i8]* %macaddr, [6 x i8]* %"@x_val"
4:         ^
```

I also tried using `macaddr->type` instead of `call.type` but got the same result

Reply:
This comes from the AssignMap codegen. It thinks its dealing with a simpel integer value so it creates a new allocation and then stores it into it. shouldBeOnStackAlready needs updating for the new type

Reply:
Aha, that did it. Thank you :)